xdsl/traits.py

```diff
@@ -264,6 +264,31 @@ def lookup_symbol(
                 return SymbolTable.lookup_symbol(o, nested_name)
         return None
 
+    @staticmethod
+    def insert_symbol_if_not_exists(symbol_table_op: Operation, symbol_op: Operation):
```

Comment:
Can you please document the function?
It's not obvious to me from the name what the function does, and what are the constraints that need to be fulfilled.

Reply:
Also, why adding `insert_symbol_if_not_exists`, and not `if_symbol_in_table` for instance?